src/prefabs/textinput.tsx

```diff
@@ -2,31 +2,337 @@ import * as React from 'react';
+              Action input variable
+              <CircleQuestion
+                color="grey500"
+                size="medium"
+                data-tip="You can use this action input variable in the action itself."
+                data-for="variable-tooltip"
+              />
+            </Label>
+            <BBTooltip
+              id="variable-tooltip"
+              place="top"
+              type="dark"
+              effect="solid"
+            />
+            <Text
+              onChange={(e): void => setVariableInput(e.target.value)}
+              color="orange"
+            />
+          </Field>
+        )}
+      </Content>
+      <Footer
+        onClose={close}
+        canSave={(propertyPath && !!name) || variableInput}
+        onSave={async (): Promise<void> => {
+          // eslint-disable-next-line no-param-reassign
+
+          structure.id = componentId;
+
+          let kind = propertyKind || 'STRING';
+          const isListProperty = kind === ('LIST' || 'list');
+
+          const arrayKind = 'HAS_MANY';
+          const integerKind = 'BELONGS_TO';
+
+          if (
+            originalPrefab.name === BettyPrefabs.MULTI_AUTO_COMPLETE &&
+            !propertyBased
+          ) {
+            kind = arrayKind;
+          }
+
+          if (
+            (originalPrefab.name === BettyPrefabs.SELECT ||
+              originalPrefab.name === BettyPrefabs.RADIO ||
+              originalPrefab.name === BettyPrefabs.AUTO_COMPLETE) &&
+            !propertyBased &&
+            !isListProperty
+          ) {
+            kind = integerKind;
+          }
+
+          const variableName = variableInput || name;
+          const result = await prepareInput(
+            actionId,
+            variableName,
+            kind,
+            propertyKind,
+          );
+
+          const newPrefab = { ...originalPrefab };
+          if (newPrefab.structure[0].type !== 'COMPONENT') {
+            throw new Error('expected Component');
+          }
+          setOption(newPrefab.structure[0], actionVariableOption, (option) => ({
+            ...option,
+            value: variableName,
+            configuration: {
+              condition: {
+                type: 'SHOW',
+                option: 'actionProperty',
+                comparator: 'EQ',
+                value: '',
+              },
+            },
+          }));
+          setOption(newPrefab.structure[0], labelOptionKey, (option) => ({
+            ...option,
+            value: [variableName],
+          }));
+
+          if (
+            originalPrefab.name === BettyPrefabs.SELECT ||
+            originalPrefab.name === BettyPrefabs.RADIO
+          ) {
+            setOption(newPrefab.structure[0], 'property', (option) => ({
+              ...option,
+              value: {
+                id: propertyId,
+                type: DynamicTypeEnum.Property,
+                componentId: selectedPrefab?.id,
+              },
+            }));
```

Comment:
DynamicTypeEnum is a unnecessary here since we only use 'STRING' here.

Also I think we should probably introduce helpers to create options so that we can save the pro-coder from having to deal with these details.

```ts
setOption(newPrefab.structure[0], 'property', (option) => ({
              ...option,
              ...propertyOption(propertyId, selectedPrefab?.id)
            }));
```

Reply:
I resolved first, but the second I think we can leave up to paradigm when they will extend it to other inputs